scripts/scanner.js

```diff
@@ -0,0 +1,75 @@
+import QrScanner from 'qr-scanner';
+
+import { doms } from './global';
+import { ALERTS } from './i18n';
+import { createAlert } from './misc';
+
+/**
+ * The active QR scanner (if one exists)
+ * @type {QrScanner}
+ */
+let scanner = null;
+
+/**
+ * Asynchronously prompt a QR code scan, returning the QR string data on resolve or a `false` rejection on cancel or error
+ * @returns {Promise<string> | Promise<false>} - QR String data | false
+ */
+export async function scanQRCode() {
+    // Don't create multiple scanners; in case of button spam
+    if (scanner) return false;
+
+    // Check for Camera support
+    if (!QrScanner.hasCamera()) {
+        createAlert('warning', ALERTS.NO_CAMERAS, [], 3000);
+    }
+
+    return new Promise((resolve, reject) => {
+        // Create a scanner
+        scanner = new QrScanner(
+            doms.domQrReaderStream,
+            (res) => {
+                stopQRScan();
+                resolve(res);
+            },
+            {
+                returnDetailedScanResult: true,
+                preferredCamera: 'environment',
+                highlightCodeOutline: true,
+                highlightScanRegion: true,
+            }
+        );
+
+        // Attempt to start scanning, then display the UI
+        try {
```

Comment:
I think try blocks don't work on promises that you don't await, we should use `.catch()` instead